mysql/README.md

```diff
@@ -1,41 +1,88 @@
 
-    # docker run -d -p 3306:3306 <yourname>/mysql
+|  Volume mount point      | Description          |
+| :----------------------- | -------------------- |
+|  `/var/lib/mysql/data`   | MySQL data directory |
 
-Get container ID:
+**Notice: When mouting a directory from the host into the container, ensure that the mounted
+directory has the appropriate permissions and that the owner and group of the directory
+matches the user UID or name which is running inside the container.**
 
-    # docker ps
+Usage
+---------------------------------
 
-Keep in mind the password set for MySQL is: mysqlPassword
+For this, we will assume that you are using the `openshift/mysql-55-centos7` image.
+If you want to set only the mandatory environment variables and not store
+the database in a host directory, execute the following command:
 
-Get the IP address for the container:
+```
+$ docker run -d --name mysql_database -e MYSQL_USER=user -e MYSQL_PASSWORD=pass -e MYSQL_DATABASE=db -p 3306:3306 fedora/mysql
+```
 
-    # docker inspect <container_id> | grep -i ipaddr
+This will create a container named `mysql_database` running MySQL with database
+`db` and user with credentials `user:pass`. Port 3306 will be exposed and mapped
+to the host. If you want your database to be persistent across container executions,
+also add a `-v /host/db/path:/var/lib/mysql/data` argument. This will be the MySQL
+data directory.
 
-For MySQL:
-    # mysql -h 172.17.0.x -utestdb -pmysqlPassword
+If the database directory is not initialized, the entrypoint script will first
+run [`mysql_install_db`](https://dev.mysql.com/doc/refman/5.6/en/mysql-install-db.html)
+and setup necessary database users and passwords. After the database is initialized,
+or if it was already present, `mysqld` is executed and will run as PID 1. You can
+ stop the detached container by running `docker stop mysql_database`.
 
 
-Create a table:
+MySQL root user
+---------------------------------
+The root user has no password set by default, only allowing local connections.
+You can set it by setting the `MYSQL_ROOT_PASSWORD` environment variable. This
+will allow you to login to the root account remotely. Local connections will
+still not require a password.
+
+To disable remote root access, simply unset `MYSQL_ROOT_PASSWORD` and restart
+the container.
+
+
+Changing passwords
+------------------
+
+Since passwords are part of the image configuration, the only supported method
+to change passwords for the database user (`MYSQL_USER`) and root user is by
+changing the environment variables `MYSQL_PASSWORD` and `MYSQL_ROOT_PASSWORD`,
+respectively.
+
+Changing database passwords through SQL statements or any way other than through
+the environment variables aforementioned will cause a mismatch between the
+values stored in the variables and the actual passwords. Whenever a database
+container starts it will reset the passwords to the values stored in the
+environment variables.
 
```

Comment:
This seems to be one of the other reasons driving complexity - the need to update the password of an already existing image.

Why do we have a password at all?

My thought was that it might be necessary in order to work with software which is opinionated or just buggy.  (Scripts that end up running something like `mysql -uroot -p ""` won't work; I think you have to use `--password=`).

But in that case I don't think you need to be able to change the password  You just initialize it to the simplest placeholder that the opinionated or buggy software will tolerate.